lib/Suggestions.js

```diff
@@ -73,35 +76,41 @@ class Suggestions extends Component {
   };
 
   shouldRenderSuggestions = (query) => {
-    const { props } = this;
-    const minQueryLength = Number.isInteger(props.minQueryLength)
-      ? props.minQueryLength
+    const { isFocused, minQueryLength } = this.props;
+    const minimumQueryLength = Number.isInteger(minQueryLength)
+      ? minQueryLength
       : 2;
-    return query.length >= minQueryLength && props.isFocused;
+    return query.length >= minimumQueryLength && isFocused;
   };
 
-  render() {
-    const { props } = this;
-    const suggestions = props.suggestions.map(
-      function(item, i) {
-        return (
-          <li
-            key={i}
-            onMouseDown={props.handleClick.bind(null, i)}
-            onMouseOver={props.handleHover.bind(null, i)}
-            className={
-              i == props.selectedIndex ? props.classNames.activeSuggestion : ''
-            }>
-            <span dangerouslySetInnerHTML={this.markIt(item, props.query)} />
-          </li>
-        );
-      }.bind(this)
+  renderSuggestion = (item, i) => {
+    const { classNames, handleClick, handleHover, query, selectedIndex, suggestionRenderer } = this.props;
+    const suggestionLabel = (
+      <span className={classNames.suggestionLabel} dangerouslySetInnerHTML={this.markIt(item, query)}/>
     );
+    const content = suggestionRenderer ? suggestionRenderer(suggestionLabel) : suggestionLabel;
```

Comment:
`suggestionRenderer ? suggestionRenderer(suggestionLabel) : (
      <span className={classNames.suggestionLabel} dangerouslySetInnerHTML={this.markIt(item, query)}/>
     );`

Reply:
`<span className={classNames.suggestionLabel} dangerouslySetInnerHTML={this.markIt(item, query)}/>` is `suggestionLabel`. The content is either or `suggestionLabel` or the result of the custom renderer to which is passed `suggestionLabel`. If I were to add the html inline, I would duplicate code.